### PR TITLE
Update rules on which events to add to `auth_events`

### PR DESCRIPTION
**Pull request by erikjohnston**

### CODE REVIEW

**Comment by turt2live** [comment]
I'm suspecting this PR or one of the others in the series fixes https://github.com/matrix-org/matrix-doc/issues/1430

**Comment by erikjohnston** [comment]
Yeah, I think this fixes #1430

File: specification/server_server_api.rst [hunk 1]
```diff
@@ -299,8 +299,15 @@ following subset of the room state:
 
 - The ``m.room.create`` event.
 - The current ``m.room.power_levels`` event, if any.
-- The current ``m.room.join_rules`` event, if any.
 - The sender's current ``m.room.member`` event, if any.
+- If type is ``m.room.member``:
+
+    - The target's current ``m.room.member`` event, if any.
+    - If ``join`` or ``invite`` then the current ``m.room.join_rules`` event,
```
Comment (richvdh) [inline specification/server_server_api.rst]: If *membership* is...

Comment (richvdh) [inline specification/server_server_api.rst]: s/then/,/

File: specification/server_server_api.rst [hunk 2]
```diff
@@ -299,8 +299,15 @@ following subset of the room state:
...
+      if any.
+    - If ``invite`` with ``third_party_invite`` in ``content``, then add
```
Comment (richvdh) [inline specification/server_server_api.rst]: if *membership* is...

"if content contains a third_party_invite property"

Comment (richvdh) [inline specification/server_server_api.rst]: s/then add/the/

File: specification/server_server_api.rst [hunk 3]
```diff
@@ -299,8 +299,15 @@ following subset of the room state:
...
+      ``m.room.third_party_invite`` with state_key of ``token`` in ``signed``
```
Comment (richvdh) [inline specification/server_server_api.rst]: I don't even understand this bit, tbh.

**Comment by erikjohnston** [comment]
Hopefully that makes me sense.

Ref for this PR: https://github.com/matrix-org/synapse/blob/74854a97191191b08101821753c2672efc2a65fd/synapse/event_auth.py#L682

File: specification/server_server_api.rst [hunk 4]
```diff
@@ -303,11 +303,12 @@ following subset of the room state:
 - If type is ``m.room.member``:
 
     - The target's current ``m.room.member`` event, if any.
-    - If ``join`` or ``invite`` then the current ``m.room.join_rules`` event,
-      if any.
-    - If ``invite`` with ``third_party_invite`` in ``content``, then add
-      ``m.room.third_party_invite`` with state_key of ``token`` in ``signed``
-      field of ``third_party_invite``, if any.
+    - If membership is ``join`` or ``invite``, the current ``m.room.join_rules``
```
Comment (richvdh) [inline specification/server_server_api.rst]: backticks on membership would have been nice

File: specification/server_server_api.rst [hunk 5]
```diff
@@ -303,11 +303,12 @@ following subset of the room state:
...
+      event, if any.
+    - If membership is ``invite`` and ``content`` contains a
+      ``third_party_invite`` property, the current
+      ``m.room.third_party_invite`` with state_key matching
```
Comment (richvdh) [inline specification/server_server_api.rst]: `m.room.third_party_invite` *event*

Comment (richvdh) [inline specification/server_server_api.rst]: backticks on state_key

**Comment by richvdh** [comment]
[yes it does make much more sense. thanks.]

**Pull request merged**